analyses/cell-type-nonETP-ALL-03/scripts/00-01_processing_rds.R

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env Rscript
+
+#This script uses SAM algorithm to perform soft feature selection for better separation of homogenous populations.
+#So we use SAM to preprocess the data, perform feature selection/dimensionality reduction and clustering.
+#The outputs are an intermediate rds file and umap plot showing leiden clustering results.
+
+library(Seurat)
+library(SingleCellExperiment)
+library(ggplot2)
+######## step01 #######################################
+library(reticulate)
+use_condaenv("openscpca-cell-type-nonETP-ALL-03")
+samalg <- import("samalg") #https://github.com/atarashansky/self-assembling-manifold/tree/master
+
+run_sam <- function(sample, library){
+  sce <- readRDS(file.path(data_loc,sample,paste0(library,"_processed.rds")))
+  seu <- CreateSeuratObject(counts = counts(sce), data = logcounts(sce), meta.data = as.data.frame(colData(sce)))
+  # create a new assay to store ADT information
+  seu[["ADT"]] <- CreateAssayObject(counts = counts(altExp(sce)))
+  seu[["ADT"]]@data <- logcounts(altExp(sce))
+  mito.features <- rownames(sce)[which(grepl("^MT-", rowData(sce)$gene_symbol))]
+  seu[["percent.mt"]] <- PercentageFeatureSet(seu, features = mito.features)
+  
+  #detecting doublets
+  doublets.data <- read.table(file = file.path(doublet_loc,sample,paste0(library,"_processed_scdblfinder.tsv")),sep = "\t", header = T)
+  idx <- match(colnames(seu), doublets.data$barcodes)
+  seu$doublet_class <- doublets.data$class[idx]
+  seu <- subset(seu, subset = percent.mt < 25)
+  
+  #step 01 feature selection/dimensionality using SAM
+  sam = samalg$SAM(counts = c(r_to_py(t(seu@assays[["RNA"]]@layers[["counts"]])), 
+                              r_to_py(as.array(rownames(seu))),
+                              r_to_py(as.array(colnames(seu)))))
+  sam$preprocess_data()
+  sam$run()
+  sam$clustering(method = "leiden") #leiden clustering is the default algorithm in SAM
+  sam$save_anndata(paste0("sam.",sample,".h5ad"))
+  
+  final.obj <- schard::h5ad2seurat(paste0("sam.",sample,".h5ad"))
+  final.obj <- AddMetaData(final.obj, seu@meta.data)
+  final.obj@assays[["RNA"]]@counts <- seu@assays[["RNA"]]@layers[["counts"]]
+  # create a new assay to store ADT information
+  final.obj[["ADT"]] <- CreateAssayObject(counts = seu@assays[["ADT"]]@counts)
+  final.obj@assays[["ADT"]]@data <- seu@assays[["ADT"]]@data
```

Comment:
Thank you for simplifying it! Since `percent.mt` and `subsets_mito_percent` are storing the same information, I only keep `subsets_mito_percent`. Just to check with you, are `nCount_RNA` and `nFeature_RNA` metadata stored in `sce` object? Because these two metadata give different values between creating seurat object via `CreateSeuratObject` (named as seu) and `schard::h5ad2seurat` (named as final.obj)

<img width="697" alt="Screenshot 2024-09-25 at 11 16 28 AM" src="https://github.com/user-attachments/assets/681283c7-f7ba-41cf-8a33-42d8461bdc77">

It seems like `CreateSeuratObject` is giving the right value, since the `min_gene_cutoff > 200`, so the minimum value of `nFeature_RNA` should be larger than 200, right?

Reply:
Thank you for the clarification! I have now set them as:
````
final.obj$nCount_RNA <- final.obj$sum
final.obj$nFeature_RNA <- final.obj$detected
````